src/wallet/walletutil.cpp

```diff
@@ -57,10 +57,16 @@ bool IsBerkeleyBtree(const fs::path& path)
 std::vector<fs::path> ListWalletDir()
 {
     const fs::path wallet_dir = GetWalletDir();
-    const size_t offset = wallet_dir.string().size() + 1;
+    size_t offset = wallet_dir.string().size();
     std::vector<fs::path> paths;
     boost::system::error_code ec;
 
+    // account for possible trailing backslash, since we have boost below 1,60
+    while(wallet_dir.string()[offset-1] == '\\' || wallet_dir.string()[offset-1] == '/')
+    {
```

Comment:
Would it be safer to use `.at()` for bounds checks/no UB?

nit, clang-formatting
```diff
-    while(wallet_dir.string()[offset-1] == '\\' || wallet_dir.string()[offset-1] == '/')
-    {
+    while (wallet_dir.string()[offset - 1] == '\\' || wallet_dir.string()[offset - 1] == '/') {
```

Reply:
yup,  draft ugly and no checks, maybe i close it, that PR is now like POC.
What we need is an elegant way with just boost 1.47   to do this. 
Ideas?